File: Mathematical Algorithms/Super Prime/SP.py
# Generate all prime numbers less than n.
def SieveOfEratosthenes(n, isPrime):
    isPrime[0] = isPrime[1] = False
    for i in range(2, n + 1):
        isPrime[i] = True
    for p in range(2, n + 1):
        if p * p <= n and isPrime[p] == True:
            for i in range(p * 2, n + 1, p):
                isPrime[i] = False
                p += 1


def superPrimes(n):
    # Generating primes using Sieve
    isPrime = [1 for i in range(n + 1)]
    SieveOfEratosthenes(n, isPrime)
    # Storing all the primes generated in a, an array primes[]
    primes = [0 for i in range(2, n + 1)]
    j = 0
    for p in range(2, n + 1):
        if isPrime[p]:
            primes[j] = p
            j += 1
    # Printing all those prime numbers that occupy prime numbered position in sequence of prime numbers.
    for k in range(j):
        if isPrime[k + 1]:
            print(primes[k], end=" ")

File: Mathematical Algorithms/Super Prime/SP.cpp
#include <iostream>
using namespace std;

// Generate all prime numbers less than n.
bool SieveOfEratosthenes(int n, bool isPrime[]) {
    isPrime[0] = isPrime[1] = false;
    for (int i = 2; i <= n; i++)
        isPrime[i] = true;
    for (int p = 2; p * p <= n; p++) {
        if (isPrime[p] == true) {
            for (int i = p * 2; i <= n; i += p)
                isPrime[i] = false;
        }
    }
}

// Prints all super primes less than or equal to n.
void superPrimes(int n) {
    // Generating primes using Sieve
    bool isPrime[n + 1];
    SieveOfEratosthenes(n, isPrime);
    // Storing all the primes generated in a
    // an array primes[]
    int primes[n + 1], j = 0;
    for (int p = 2; p <= n; p++)
        if (isPrime[p])
            primes[j++] = p;
    // Printing all those prime numbers that occupy prime numbered position in sequence of prime numbers.
    for (int k = 0; k < j; k++)
        if (isPrime[k + 1])
            cout << primes[k] << " ";
}